ChatSX/screens/ChatRoom/index.tsx

```diff
@@ -2426,6 +2786,122 @@ const ChatRoom = ({navigation, route}: ChatRoom) => {
         </View>
       ) : (
         <>
+          {!(Object.keys(currentChatroomTopic).length === 0) &&
+          chatroomType !== ChatroomType.DMCHATROOM ? (
+            <Pressable
+              onPress={async () => {
+                let index = conversations.findIndex(
+                  (element: any) => element?.id === currentChatroomTopic?.id,
+                );
+                if (index >= 0) {
+                  flatlistRef.current?.scrollToIndex({
+                    animated: true,
+                    index,
+                  });
+                  setIsFound(true);
+                } else {
+                  const topicConversation = await myClient?.getConversation(
+                    currentChatroomTopic?.id,
+                  );
+                  let payload = {
+                    type: GetConversationsType.ABOVE,
+                    limit: 100,
+                    medianConversation: currentChatroomTopic,
+                  };
+                  const aboveConversations =
+                    await myClient?.getPaginatedConversations(payload);
+                  payload.type = GetConversationsType.BELOW;
+                  const belowConversations =
+                    await myClient?.getPaginatedConversations(payload);
+                  let newConversation = aboveConversations.concat(
+                    topicConversation,
+                    belowConversations,
+                  );
+                  newConversation = newConversation.reverse();
+
+                  dispatch({
+                    type: GET_CONVERSATIONS_SUCCESS,
+                    body: {conversations: newConversation},
+                  });
+                  let index = newConversation.findIndex(
+                    (element: any) => element?.id === currentChatroomTopic?.id,
+                  );
+                  if (index >= 0) {
+                    flatlistRef.current?.scrollToIndex({
+                      animated: true,
+                      index,
+                    });
+                    setIsFound(true);
+                  }
+                }
```

Comment:
create a function for `onPress()` as it will used on other places as well, please check ED for this

Reply:
Tried creating a method named `scrollToConversationIndex` in chatroom utils file but not able to do so as this method contains hook state like `setIsFound`, `dispatch` which is `useAppDispatch` hook instance, etc and all this can be used only inside a functional component otherwise it throws error. Though will divide this method into as many smaller methods possible.